### ISSUE TITLE
Cache data in a database

### ISSUE THREAD

**Issue by barnesjt** [Barnesjt/appstudio-weather#5]
The weather data must be stored in some kind of database, which can be used if the device is offline.

**Comment by barnesjt** [comment]
Relevant documentation:
https://developers.arcgis.com/appstudio/api-reference/qml-arcgis-appframework-sql-sqldatabase

For storing local data, I looked at an SQLite database. The setup is fairly straightforward. The schema here is not set in stone, it just stores the response json as a string, which works, but just pushes back the parsing work that needs to be done anyways. I think it would be best to do the work upfront.

Here's the code I wrote yesterday while testing my understanding:

```
Item {
    id: weatherDB

    property real longitude
    property real latitude
    property real time: 0
    property var jsonObj
    property string jsonString: "nothing yet"


    FileFolder {
      id: dbFilePath
      path: "~/ArcGIS/Data/Sql"
    }

    SqlDatabase {
      id: db
      databaseName: dbFilePath.filePath("weather.sqlite")
    }

    function getDBRecent() {
      db.open();
      var query = db.query("SELECT * FROM weather_cache;");
      query.first();
      weatherDB.jsonString = query.values.strRes;
      weatherDB.longitude = query.values.longitude;
      weatherDB.latitude = query.values.latitude;
      weatherDB.jsonObj = JSON.parse(weatherDB.jsonString);
      weatherDB.time = query.values.time;
      db.close();
    }

    function putDB(time, lat, lon, res) {
      db.open();
      var query = db.query("DELETE FROM weather_cache");
      var insertString = "INSERT INTO weather_cache (time, latitude, longitude, strRes) VALUES ('"+time+"','"+lat+"','"+lon+"','"+res+"')";
      db.query(insertString);
      db.close();
      
    }

    Component.onCompleted: {
      dbFilePath.makeFolder();
      db.open();
      db.query("CREATE TABLE IF NOT EXISTS weather_cache (id INTEGER PRIMARY KEY AUTOINCREMENT, time INTEGER NOT NULL, latitude FLOAT NOT NULL, longitude FLOAT NOT NULL, strRes TEXT NOT NULL)");
      db.close();
      getDBRecent();
      if(Date.now() - weatherDB.time > 60000){
        weatherRequest.send();
      }
    }
```

Queries and schema are just inlined, for now, I imagine there is a better mechanism for this, but I'm not aware of it presently. In the implementation of this app, I would probably rework the logic a bit, but this does work as is.

**Comment by barnesjt** [comment]
Resolved in PR #9
